docs/tutorials/ptypy-workflows.md

```diff
@@ -0,0 +1,440 @@
+    COPY ./cufft/dependencies.yml ./dependencies_cufft.yml
+    RUN mamba install cuda-version=${CUDAVERSION} && \
+        mamba env update -n base -f dependencies.yml && \
+        mamba env update -n base -f dependencies_cufft.yml
+
+    # Pull from MPI build and install accelerate/cupy dependencies
+    FROM mpi as cupy
+    ARG CUDAVERSION
+    COPY ./ptypy/accelerate/cuda_cupy/dependencies.yml ./dependencies.yml
+    COPY ./cufft/dependencies.yml ./dependencies_cufft.yml
+    RUN mamba install cuda-version=${CUDAVERSION} && \
+        mamba env update -n base -f dependencies.yml && \
+        mamba env update -n base -f dependencies_cufft.yml
+
+    # Pull from platform specific image and install ptypy 
+    FROM ${PLATFORM} as build
+    COPY pyproject.toml setup.py ./
+    COPY ./scripts ./scripts
+    COPY ./templates ./templates
+    COPY ./benchmark ./benchmark
+    COPY ./cufft ./cufft
+    COPY ./ptypy ./ptypy
+    RUN pip install .
+
+    # For core/full build, no post processing needed
+    FROM build as core-post
+    FROM build as full-post
+
+    # For pycuda build, install filtered cufft
+    FROM build as pycuda-post
+    RUN pip install ./cufft
+
+    # For pycuda build, install filtered cufft
+    FROM build as cupy-post
+    RUN pip install ./cufft
+
+    # Platform specific runtime container
+    FROM ${PLATFORM}-post as runtime
+
+    # Run PtyPy run script as entrypoint
+    ENTRYPOINT ["ptypy.run"]
+    ```
+can be used to build a container image with openmpi as the MPI backend and CuPy as the backend for PtyPy
+
+```bash
+podman build . -t gcr.io/diamond-privreg/ptypy/test_openmpi_cupy:0.1 --target=runtime --build-arg MPI=openmpi --build-arg PLATFORM=cupy
+```
+
+and then be pushed to Diamond's the private container registry
+
+```bash
+podman push gcr.io/diamond-privreg/ptypy/test_openmpi_cupy:0.1
+```
+
+## Cluster Workflow Templates
+
+The next step is to create a ```ClusterWorkFlowTemplate``` and push it to the corresponding GitHub
+repository for science workflows, in this case [imaging-workflows](https://github.com/DiamondLightSource/imaging-workflows).
+The Argo workflow engine can be configured to automatically look for existing templates in those
+repositories and make them available via the [dashboard](https://workflows.diamond.ac.uk) or the graph (once available).
+
+!!! note "Workflow template repositories"
+    If there is currently no workflow repository for your science area, send a message to a
+    member of the analysis platform team and we can get it sorted and configure the workflow
+    engine to automatically pick up your workflow templates from the new repository.
+
+## Writing a Cluster Workflow Template
+
+Cluster Workflow Templates are largely the same as [Workflow Templates](../how-tos/submit-workflow.md)
+with a basic structure looking like this
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: ClusterWorkflowTemplate
+metadata:
+  name: ptypy-cpu-job
```

Comment:
Perhaps better to use `generateName` here?

Reply:
not sure if `generateName` was working in a ClusterWorkflowTemplate. Can check once all the CRDs are available